File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

let [N, M] = input[0].split(' ').map(el => +el);
let basket = [];
for(let i=1; i<=N; i++){
    basket.push(i);
};

for(let i=1; i<=M; i++){
    let [a, b] = input[i].split(' ').map(el => +el);
    let arr = [];
    for(let j=a-1; j<b; j++){
        arr.push(basket[j])
    };
    arr.reverse();
    basket.splice(a-1, b-a+1, ...arr);
}
console.log(basket.join(' '))

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
N, M = map(int, input().split())
numbers = []

for i in range(1, N + 1):
    numbers.append(i)

for i in range(M):
    j, k = map(int, input().split())
    temp = numbers[j - 1:k]
    temp.reverse()
    numbers[j - 1:k] = temp

for i in range(N):
    print(numbers[i], end=" ")
